packages/test/test-end-to-end-tests/src/test/batchBreakRegression.spec.ts

```diff
@@ -45,7 +45,10 @@ type ProxyOverrides<T> = {
 		: OverrideFunction<T, P>;
 };
 
-function createFunctionOverrideProxy<T extends object>(obj: T, overrides: ProxyOverrides<T>): T {
+function createFunctionOverrideProxy<T extends Record<string, any>>(
+	obj: T,
+	overrides: ProxyOverrides<T>,
+): T {
```

Comment:
Changing 'extends object' to 'extends Record<string, any>' fixed inference for 'typeof foo === "function"'.